Microsoft.DotNet.Interactive/KernelHierarchy.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Interactive
+{
+    internal static class KernelHierarchy
```

Comment:
Are you sure we need this? It's typically going to be more robust to traverse the tree on demand (which sending a command into the root kernel already does) rather than maintain a separate representation of it.

Reply:
That class is used to track the root kernel